.github/workflows/tests.yml

```diff
@@ -0,0 +1,423 @@
+        # We cannot 'uses: pypa/gh-action-pypi-publish@v1.4.1' because
+        # that's apparently a container action, and those don't run on
+        # the Mac.
+        if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags') && startsWith(runner.os, 'Mac')
+        env:
+          TWINE_PASSWORD: ${{ secrets.TWINE_PASSWORD }}
+        run: |
+          twine upload --skip-existing dist/*
+
+  test:
+    needs: build-zope_interface
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        python-version: [2.7, pypy-2.7, pypy-3.6, 3.5, 3.6, 3.7, 3.8, 3.9]
+        os: [ubuntu-20.04, macos-latest]
+        exclude:
+          - os: macos-latest
+            python-version: pypy-2.7
+          - os: macos-latest
+            python-version: pypy-3.6
+          - os: macos-latest
+            python-version: 3.5
+
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Get pip cache dir
+        id: pip-cache
+        run: |
+          echo "::set-output name=dir::$(pip cache dir)"
+
+      - name: pip cache
+        uses: actions/cache@v2
+        with:
+          path: ${{ steps.pip-cache.outputs.dir }}
+          key: ${{ runner.os }}-pip-${{ matrix.python-version }}
+          restore-keys: |
+            ${{ runner.os }}-pip-
+
+      - name: Download zope.interface wheel
+        uses: actions/download-artifact@v2
+        with:
+          name: zope.interface-${{ runner.os }}-${{ matrix.python-version }}.whl
+          path: dist/
+      - name: Install zope.interface
+        # ``python -m unittest discover`` only works with editable
+        # installs, so we have to duplicate some work and can't
+        # install the built wheel. (zope.testrunner
+        # works fine with non-editable installs.)
+        run: |
+          pip install -U wheel
+          pip install -U coverage
+          pip install -U 'faulthandler; python_version == "2.7" and platform_python_implementation == "CPython"'
+          # Unzip into src/ so that testrunner can find the .so files
+          # when we ask it to load tests from that directory. This
+          # might also save some build time?
+          unzip -n dist/zope.interface-*whl -d src
+          pip install -U -e .[test]
+      - name: Run tests and report coverage
+        # Once with C extensions, once without. Yes, this runs them
+        # twice on PyPy.
+        run: |
+          coverage run -p -m unittest discover -s src
+          PURE_PYTHON=1 coverage run -p -m unittest discover -s src
+          coverage combine
+          coverage report -i
+      - name: Submit to Coveralls
+        # This is a container action, which only runs on Linux.
+        if: ${{ startsWith(runner.os, 'Linux') }}
+        uses: AndreMiras/coveralls-python-action@develop
```

Comment:
Is it a good idea to run bleeding edge action code instead of a fixed tag?

Reply:
Great point. That's what [the documentation](https://github.com/AndreMiras/coveralls-python-action) and other examples I've seen have done, so that's what I've been doing with this action too. 

We could try pinning to a tag; the newest tag is `v20201129`, making it three months old. If we did that, though, we'd probably also want to pin `coverage` to the version that was current at that time, wouldn't we? And I'm not sure we'd want to do that. Since the action also includes `coveralls`, we'd be pinning ourself to an older version of that, something we also don't usually do IIRC; I'd be concerned about the coveralls API changing.  In the past, IIRC, coveralls submission failures have been basically silent and hard to debug.

I'm not sure which is the worst of the two.

Reply:
Well, if the docs say so, I suppose it's fine.